relays/messages/src/relay_strategy/rational_strategy.rs

```diff
@@ -205,47 +179,20 @@ impl RelayStrategy for RationalStrategy {
 				selected_cost = total_cost;
 			}
 
-			hard_selected_count = index + 1;
-			selected_weight = new_selected_weight;
-			selected_unpaid_weight = new_selected_unpaid_weight;
-			selected_prepaid_nonces = new_selected_prepaid_nonces;
 			selected_size = new_selected_size;
 			selected_count = new_selected_count;
 		}
 
-		if hard_selected_count != soft_selected_count {
-			let hard_selected_end_nonce =
-				hard_selected_begin_nonce + hard_selected_count as MessageNonce - 1;
-			let soft_selected_begin_nonce = hard_selected_begin_nonce;
-			let soft_selected_end_nonce =
-				soft_selected_begin_nonce + soft_selected_count as MessageNonce - 1;
-			log::warn!(
-				target: "bridge",
-				"Relayer may deliver nonces [{:?}; {:?}], but because of its strategy (Rational) it has selected \
-				nonces [{:?}; {:?}].",
-				hard_selected_begin_nonce,
-				hard_selected_end_nonce,
-				soft_selected_begin_nonce,
-				soft_selected_end_nonce,
-			);
-
-			hard_selected_count = soft_selected_count;
-		}
-
-		if hard_selected_count != 0 {
+		if soft_selected_count != 0 {
 			log::trace!(
 				target: "bridge",
-				"Expected reward from delivering nonces [{:?}; {:?}] is: {:?} - {:?} = {:?}",
+				"Expected reward from delivering nonces [{:?}; ?] is: {:?} - {:?} = {:?}",
```

Comment:
About there, because can not get `hard_selected_count`, it's in `EnforcementStrategy`, maybe we can add a new parameter to the decide function?